utilities/hooks.py

```
@@ -64,29 +64,39 @@ def validate_hook_config(hook_config: dict[str, Any], hook_type: str) -> None:
+    modes_specified = sum(x is not None for x in (expected_result, custom_playbook, aap_job_template_id))
 
-    # Validate mutual exclusivity
-    if expected_result is not None and custom_playbook is not None:
+    if modes_specified > 1:
         raise ValueError(
-            f"Invalid {hook_type} hook config: 'expected_result' and 'playbook_base64' are "
-            f"mutually exclusive. Use 'expected_result' for predefined playbooks, or "
-            f"'playbook_base64' for custom playbooks."
+            f"Invalid {hook_type} hook config: 'expected_result', 'playbook_base64', and "
+            f"'aap_job_template_id' are mutually exclusive. Specify exactly one."
         )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)